src/pages/ProfilesExplorerView/infrastructure/labels/LabelsDataSource.ts

```diff
@@ -82,22 +87,17 @@ export class LabelsDataSource extends RuntimeDataSource {
       return [];
     }
 
-    // we could interpolate ad hoc filters, but the Service labels exploration type would reload all labels each time they are modified
-    // const filters = sceneGraph.interpolate(sceneObject, '$filters');
-    // const pyroscopeQuery = `${profileMetricId}{service_name="${serviceName}",${filters}}`;
-    const pyroscopeQuery = `${profileMetricId}{service_name="${serviceName}"}`;
-
-    const { from, to } = computeRoundedTimeRange(range as TimeRange);
+    labelsRepository.setApiClient(new LabelsApiClient({ dataSourceUid }));
```

Comment:
The fix. We don't cache the API client anymore with `DataSourceProxyClientBuilder`, so that when the query builder aborts a request, it will do it via its own API client.